### PR TITLE
#400 RsPrettyXML causes NPE

### LINKED ISSUE #400: RsPrettyXML causes NPE
The code below causes null pointer exception
I believe the issue has to be fixed
```
new RsPrettyXML(
    new RsWithBody(                        
        "<!DOCTYPE html SYSTEM \"about:legacy-compat\"><html><head></head><body></body></html>"
    )
)
```
>java.lang.NullPointerException
	at java.util.Hashtable.put(Hashtable.java:459)
	at java.util.Properties.setProperty(Properties.java:166)
	at net.sf.saxon.jaxp.IdentityTransformer.setOutputProperty(IdentityTransformer.java:293)
	at org.takes.rs.RsPrettyXML.prepareDocType(RsPrettyXML.java:171)
	at org.takes.rs.RsPrettyXML.transform(RsPrettyXML.java:135)
	at org.takes.rs.RsPrettyXML.make(RsPrettyXML.java:105)
	at org.takes.rs.RsPrettyXML.body(RsPrettyXML.java:91)
	at org.takes.rs.RsWrap.body(RsWrap.java:63)
	at org.takes.rs.RsPrint.printBody(RsPrint.java:157)
	at org.takes.rs.RsPrint.printBody(RsPrint.java:88)
	at org.takes.rs.RsPrettyXMLTest.formatsHtml5DoctypeBody(RsPrettyXMLTest.java:74)

### PR DESCRIPTION
**Pull request by xupyprmv**
I added support of empty doctype PUBLIC and empty doctype SYSTEM to solve #400.

### CODE REVIEW

**Comment by davvd** [comment]
@xupyprmv Thanks, let me find someone who can review this pull request

**Comment by davvd** [comment]
@dmzaytsev review this one,please

**Comment by dmzaytsev** [comment]
@xupyprmv according the quality rules http://at.teamed.io/qa.html

>Pull request title explains the problem it is fixing.
Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

**Comment by xupyprmv** [comment]
@dmzaytsev I changed title and description of PR.

File: src/main/java/org/takes/rs/RsPrettyXML.java [hunk 1]
```diff
@@ -165,11 +165,15 @@ private static void prepareDocType(final InputStream body,
                     && "html".equalsIgnoreCase(doctype.getName())) {
                     transformer.setOutputProperty(OutputKeys.METHOD, "html");
                     transformer.setOutputProperty(OutputKeys.VERSION, "5.0");
-                } else {
```
Comment (dmzaytsev) [inline src/main/java/org/takes/rs/RsPrettyXML.java]: @xupyprmv I'm not sure, should we set "method" property only when publicId and systemId is null ?

Comment (dmzaytsev) [inline src/main/java/org/takes/rs/RsPrettyXML.java]: @xupyprmv never mind it's not your changes

File: src/main/java/org/takes/rs/RsPrettyXML.java [hunk 2]
```diff
@@ -165,11 +165,15 @@ private static void prepareDocType(final InputStream body,
...
+                    return;
```
Comment (dmzaytsev) [inline src/main/java/org/takes/rs/RsPrettyXML.java]: I think it would more readable to use `else` instead of `return`

**Comment by dmzaytsev** [comment]
@xupyprmv 
> @dmzaytsev I changed title and description of PR.

thanks

please see one comment above

File: src/main/java/org/takes/rs/RsPrettyXML.java (hunk 2 above)
Comment (xupyprmv) [inline src/main/java/org/takes/rs/RsPrettyXML.java]: Look at  my 2nd commit. This part of code already refactored according to checkstyle rules.

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@ypshenychka please, check this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by ypshenychka** [comment]
@dmzaytsev 
>I think it would more readable to use else instead of return

According to our [QA Rules](http://www.teamed.io/qa.html):
>Messages in a ticket always start with a name of a user they are addressed to.

Please correct your code review comment by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@dmzaytsev 
According to our [QA Rules](http://www.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.

No issues were found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by dmzaytsev** [comment]
@ypshenychka I confirm
the comment updated as well

**Comment by ypshenychka** [comment]
@dmzaytsev Thank you.

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@dmzaytsev 10 mins was added to the account of @ypshenychka (for QA review), in transaction 72548408. Much obliged! I have added **22 mins** to your account in payment "72548434", 86 hours and 7 mins spent. you're getting extra minutes here (c=7). +22 to your rating, your total score is [+1018](http://www.netbout.com/b/35166?open=rating)

**Pull request merged**